### PR TITLE
[5.5] Execute garbage collection before over limit memory queue worker stop…

### PR DESCRIPTION
**Pull request by sh-ogawa**
…ped.(laravel/ideas#1380)

### CODE REVIEW

**Comment by mfn** [comment]
I don't understand this PR with regards to the linked issue https://github.com/laravel/ideas/issues/1380

If you're already hitting the memory limit threshold, the PHP process/worker is _exiting_ => so calling `gc_collect_cycles()` IMHO is pointless.

File: src/Illuminate/Queue/Worker.php [hunk 1]
```diff
@@ -200,7 +200,10 @@ protected function stopIfNecessary(WorkerOptions $options, $lastRestart)
         }
 
         if ($this->memoryExceeded($options->memory)) {
-            $this->stop(12);
+            gc_collect_cycles();
```
Comment (GrahamCampbell) [inline src/Illuminate/Queue/Worker.php]: We might want to check if this function is available to call on the system before calling it.

**Comment by mfn** [comment]
> Just call the garbage collector before the existing if.

Please also read my concern at https://github.com/laravel/ideas/issues/1380#issuecomment-449743055 , specifically

> I think that Laravels worker could benefit from calling gc_collect_cycles() but one first have to proof someone it has no ill effects.
>
> I.e. performance might be affected when a worker runs many many small/fast jobs and thus calls gc_collect_cycles() often, thus redundantly. Maybe it should be able to be tuned (only trigger once in a while) or turned off/on. But asking these question already shows it needs more preparation to land this properly.

**Comment by deleugpn** [comment]
I don't think the performance hit of calling the garbage collector after each job is relevant.
Seem to me premature optimization.

**Comment by laurencei** [comment]
What's so bad about letting the queue worker die and restart in this example anyway? If the memory is becoming a problem, then let the process die and start again.

If your having consistent issues with memory overruns, then perhaps you should use `queue:listen` which is designed to cycle after each job.

File: src/Illuminate/Queue/Worker.php (hunk 1 above)
Comment (GrahamCampbell) [reply src/Illuminate/Queue/Worker.php]: This function may be disabled on peoples installs, so this patch release might totally break their app.

Comment (sh-ogawa) [reply src/Illuminate/Queue/Worker.php]: I see. We will check and consider it.

**Pull request closed without merging**